modules/bidResponseFilter/index.js

```diff
@@ -0,0 +1,39 @@
+import { config } from '../../src/config.js';
+import { getHook } from '../../src/hook.js';
+
+export const MODULE_NAME = 'bidResponseFilter';
+export const PUBLISHER_FILTER_REJECTION_REASON = 'PUBLISHER_FILTER_REJECTION_REASON';
+export const BID_CATEGORY_REJECTION_REASON = 'BID_CATEGORY_REJECTION_REASON';
+export const BID_ADV_DOMAINS_REJECTION_REASON = 'BID_ADV_DOMAINS_REJECTION_REASON';
+export const BID_ATTR_REJECTION_REASON = 'BID_ATTR_REJECTION_REASON';
+
+function init() {
+  getHook('addBidResponse').before(addBidResponseHook);
+};
+
+export function addBidResponseHook(next, adUnitCode, bid, reject) {
+  const { bcat = [], badv = [], battr = [] } = config.getAnyConfig('ortb2') || {};
```

Comment:
ortb2 can be different for each bid response. You can get a specific response's with `auctionManager.index.getBidderRequest(bid)?.ortb2`; however, responses can also not be associated with any request, in which case the best avialable `ortb2` is the one associated with the auction (like you used in https://github.com/prebid/Prebid.js/pull/11946)

It may be worth to add a `getOrtb2` method to `auctionManager.idex` -  similar to [getMediaTypes](https://github.com/prebid/Prebid.js/blob/95c8118583eb0737409e41c8f436f90bc134df32/src/auctionIndex.js#L35-L47), which also has a fallback for when there is not request.

Reply:
I just noticed that `battr` is not here - it's [repeated for each media type in imp](https://github.com/InteractiveAdvertisingBureau/openrtb2.x/blob/main/2.6.md). It's not clear where the publisher should set it - candidates are `ortb2Imp[mediaType].battr` and `mediaTypes[mediaType].battr`

Assuming the first one should have priority, we could:

- update [ad unit validation](https://github.com/prebid/Prebid.js/blob/master/src/prebid.js#L230-L242) for each media type to find `battr` from `ortb2Imp[type].battr`, falling back to `mediaTypes[type].battr`, and copy it to both places 
- add some unit test for ortbConverter [banner](https://github.com/prebid/Prebid.js/blob/master/test/spec/ortbConverter/banner_spec.js),  [video](https://github.com/prebid/Prebid.js/blob/master/test/spec/ortbConverter/video_spec.js) and [native](https://github.com/prebid/Prebid.js/blob/master/test/spec/ortbConverter/native_spec.js) to make sure the "right" battr wins
- this should then look for `battr` from `index.getBidRequest(bid).ortb2Imp[bid.mediaType].battr`, falling back to `index.getAdUnit(bid).ortb2Imp[bid.mediaType].battr` when there is no request for the bid

(pending confirmation on the relative priority of `ortb2Imp` vs `mediaTypes`)